samples/VanillaJSTestApp/app/default/test/browser.spec.ts

```diff
@@ -2,9 +2,31 @@ import * as Mocha from "mocha";
 import puppeteer from "puppeteer";
 import { expect } from "chai";
 import fs from "fs";
+import { TestCredential } from "../../../e2eTests/TestCredential";
 
-const SCREENSHOT_BASE_FOLDER_NAME = "./test/screenshots";
+const SCREENSHOT_BASE_FOLDER_NAME = `${__dirname}/screenshots`;
 let SCREENSHOT_NUM = 0;
+let username = "";
+let accountPwd = "";
+
+function setupScreenshotDir() {
+    if (!fs.existsSync(`${SCREENSHOT_BASE_FOLDER_NAME}`)) {
+        fs.mkdirSync(SCREENSHOT_BASE_FOLDER_NAME);
+    }
+}
+
+async function setupCredentials() {
+    const testCreds = new TestCredential();
+    const envResponse = await testCreds.getUserVarsByCloudEnvironment("azurecloud");
+    const testEnv = envResponse[0];
+    if (testEnv.upn) {
+        username = testEnv.upn;
+    }
```

Comment:
If lab response doesn't contain upn, maybe log and throw?

Reply:
I'm going to enhance some of these test utilities in a future PR. for now just getting e2e working.